lib/ibandit/iban.rb

```diff
@@ -276,19 +276,47 @@ def supports_iban_determination?
     def valid_swedish_details?
       return true unless country_code == 'SE'
 
-      bank_details = {
-        bank_code: swift_bank_code,
-        account_number: swift_account_number
-      }
+      if branch_code
```

Comment:
There's an odd split of logic here - some of the validation logic is in the `IBAN` class, some is in the `SwedishDetailsValidator` class.  Given that the `IBAN` class is getting pretty big and has lots of responsibilities, feels like we should just be able to do:

```
def valid_swedish_details?
  return true unless country_code == 'SE'
  SwedishDetailsValidator.valid?(...)
end
```